README.md

```diff
@@ -1,122 +1,128 @@
+```
 
-ruleset.outcome "I matched!"
+If no outcomes match, the default one is returned.
+``` ruby
+gifter.call(day_of_week: :friday, amount: 99)
 
-ruleset.(age: 20) #=> "I matched!"
-ruleset.(age: 20, school: "University of Alabama" #=> "I matched!"
-ruleset.(age: 16, school: 'University of Texas' #=> "I matched!"
-ruleset.(age: 10, school: 'University of Alabama') #=> nil
+# => 3
 ```
 
-## Concepts
+If conditions are not met, the fallback value is returned.
+``` ruby
+gifter.call(day_of_week: :monday, amount: 124)
 
-Ruy at its core is about evaluating something against a rule. What's a rule? And what's "something"? Let's take a look.
-
-### Rules
+# => 0
+```
 
-A `Rule` is a set of conditions. Ruy has a number of builtin conditions:
+## Key concepts
 
-* Equal (**:eq**)
-* Assert (**:assert**)
-* Between (**:between**)
-* Not (**:except**)
-* Inclusion (**:include** and **:included**)
-* Less than/Greater than (**:less\_than**, **:less\_than\_or\_equal\_to**, **:greater\_than**, **:greater\_than\_or\_equal\_to**)
+Ruy at its core is about evaluating a set of conditions against a context in order to return a result.
 
-Rules can also group conditions arbitrarily as logical ANDs (**:all**) or OR (**:any**).
+### Conditions
 
-#### RuleSet-level lazy values
+A condition evaluates the state of the context.
 
-Rules can have lazy values set for them. The context must provide a proc which is evaluted only once the first time the value is needed to evaluate a rule. The result returned by the proc application is memoized and used to evaluate subsequent rules.
+Available conditions:
 
-```ruby
-rule.let :some_expensive_calculation
-```
-#### Applying Rules
+ - all *All of the nested conditions must suffice*
+ - any *At least one of its nested conditions must suffice*
+ - assert *A context value must be truish*
+ - between *Evaluates that a context value must belong to a specified range*
+ - cond *At least one slice of two nested conditions must suffice*
+ - day_of_week *Evaluates that a Date/DateTime/Time weekday is matched*
+ - eq *Tests a context value for equality*
+ - except *Evalutes that a any of the nested conditions don't suffice*
+ - greater_than *Tests that context value is greater than something*
+ - greater_than_or_equal *Tests that context value is greater or equal than something*
+ - in *A context value must belong to a specified list of values*
+ - in_cyclic_order *TBD*
+ - include *A context value must include a specified one*
+ - less_than_or_equal *Tests that context value is less than or equal to something*
+ - less_than *Tests that context value is less than something*
 
-Rules have a `#call` method that returns true when all conditions are met. `#call` takes one argument, a `Context`, that has values for the various conditions to be evaluated against. If all the conditions pass, `#call` returns `true` and otherwise `false`. Example:
+Conditions can be nested, in such case, to the condition be sufficed, the nested conditions must
```

Comment:
Conditions can be nested. In such case, for the nesting condition to be met, the nested conditions must also be met.